src/apps/service/apps/de/listings.clj

```diff
@@ -40,17 +40,20 @@
    (augment-listing-params params short-username (perms-client/load-app-permissions short-username))))
 
 (defn- augment-search-params
-  [search_term params short-username app-ids]
-  (let [category-attrs (set (workspace-metadata-category-attrs))]
-    (if (or (empty? app-ids) (empty? category-attrs))
-      params
-      (assoc params :pre-matched-app-ids
-                    (metadata-client/filter-targets-by-ontology-search short-username
-                                                                       (categorization/get-active-hierarchy-version)
-                                                                       category-attrs
-                                                                       search_term
-                                                                       ["app"]
-                                                                       app-ids)))))
+  [search_term {:keys [app-ids public-app-ids] :as params} short-username admin?]
+  (let [category-attrs (set (workspace-metadata-category-attrs))
+        app-ids        (if admin? public-app-ids app-ids)]
+    (-> params
+        (assoc :app-ids             app-ids
```

Comment:
The app search query in `kameleon` only searches apps with IDs in this `app-ids` list.
By setting the `app-ids` param to the list of `public-app-ids`, admin searches are restricted to only public apps.